packages/gestalt/src/Masonry/dynamicHeightsUtils.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * Util functions used to update positions when an item changes the height dynamically
+ */
+import { Cache } from './Cache';
+import { ColumnSpanConfig } from './multiColumnLayout';
+import { Position } from './types';
+
+function shouldExtendArea(area: { top: number; left: number; right: number }, position: Position) {
+  return position.left < area.left || position.left + position.width > area.right;
+}
+
+function isBelowArea(area: { top: number; left: number; right: number }, position: Position) {
+  return (
+    position.left < area.right &&
+    position.left + position.width > area.left &&
+    position.top > area.top
+  );
+}
+
+function recalcHeights<T>({
+  items,
+  changedItem,
+  changedItemPosition,
+  newHeight,
+  positionStore,
+  measurementStore,
+  getColumnSpanConfig,
+}: {
+  items: ReadonlyArray<T>;
+  changedItem: T;
+  changedItemPosition: Position;
+  newHeight: number;
+  positionStore: Cache<T, Position>;
+  measurementStore: Cache<T, number>;
+  getColumnSpanConfig?: (item: T) => ColumnSpanConfig;
+}) {
+  measurementStore.set(changedItem, newHeight);
+
+  const { top, left, width, height } = changedItemPosition;
+
+  const itemsWithPositions = items.filter((item) => {
+    const position = positionStore.get(item);
+    return position && position.top > changedItemPosition.top;
```

Comment:
should this be `position.top > (changedItemPosition.top + changedItemPosition.height)`? otherwise you'd also be factoring in items which are not below the changed item